tsl/src/continuous_aggs/common.c

```diff
@@ -582,21 +582,29 @@ get_bucket_width(CAggTimebucketInfo bucket_info)
 			break;
 		case INTERVALOID:
 		{
+			/*
+			 * Original interval should not be changed, hence create a local copy
+			 * for this check.
+			 */
+			Interval interval = {
```

Comment:
nit: probably could do just `Interval interval = *bucket_info.interval`

Reply:
Ok, at least this is know known.

nit: there are potential performance impact if those functions are executed frequently, because you will have to copy arguments to each call

Reply:
In this case performance is not a big problem because this code is executed once when create a continuous aggregate so unless a "crazy" user decide to create "gazillions" of caggs concurrently we are fine here. ;-)